core/sds-aspect-model-document-generators/src/main/java/io/openmanufacturing/sds/aspectmodel/generator/docu/AspectModelDocumentationGenerator.java

```diff
@@ -13,320 +13,241 @@
+        engineConfiguration.put("event_handler.reference_insertion.class", "org.apache.velocity.app.event.implement.EscapeHtmlReference");
+        engineConfiguration.put("velocimacro.library",
+                DOCU_TEMPLATE_ROOT_DIR + "/html/characteristic-documentation-lib.vm," +
+                        DOCU_TEMPLATE_ROOT_DIR + "/html/constraint-documentation-lib.vm," +
+                        DOCU_TEMPLATE_ROOT_DIR + "/html/diagram-documentation-lib.vm," +
+                        DOCU_TEMPLATE_ROOT_DIR + "/html/entity-documentation-lib.vm," +
+                        DOCU_TEMPLATE_ROOT_DIR + "/html/operation-documentation-lib.vm," +
+                        DOCU_TEMPLATE_ROOT_DIR + "/html/property-documentation-lib.vm," +
+                        DOCU_TEMPLATE_ROOT_DIR + "/html/common-documentation-lib.vm");
+
+        final Predicate<Locale> byLanguage = locale -> selectedLanguage == null || locale.getLanguage().equals(
+                selectedLanguage.getLanguage());
+        languages.stream().filter(byLanguage).forEach(language -> {
+
+            logMissingTranslations(aspect, language);
+
+            context.put("i18n", new I18nLanguageBundle(language));
+            final TemplateEngine templateEngine = new TemplateEngine(context, engineConfiguration);
+
+            final String artifactName = getArtifactName(aspect, language);
+            if (nameMapper instanceof BufferingNameMapper) {
+                ((BufferingNameMapper) nameMapper).setLanguageForArtifact(artifactName, language);
+            }
+
+            try (final OutputStream outputStream = nameMapper
+                    .apply(format.getArtifactFilename(artifactName))) {
+
+                final String generatedSource = templateEngine
+                        .apply(DOCU_ROOT_DIR + "/templates/html/aspect-model-documentation");
+
+                writeCharSequenceToOutputStream(generatedSource, outputStream);
+            } catch (final IOException e) {
+                LOG.error("Failure on writing generated Asciidoc", e);
+            }
+        });
+    }
+
+    public String getArtifactName(final Aspect aspectModel, final Locale locale) {
+        return aspectModel.getName() + "_" + locale.toLanguageTag();
+    }
+
+    private static class BufferingNameMapper implements Function<String, OutputStream> {
+        private final Map<String, ByteArrayOutputStream> artifacts = new HashMap<>();
+        private final Map<String, Locale> artifactLanguages = new HashMap<>();
+
+        public void setLanguageForArtifact(final String artifactName, final Locale language) {
+            artifactLanguages.put(artifactName, language);
+        }
+
+        @Override
+        public OutputStream apply(final String artifactName) {
+            final ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+            artifacts.put(artifactName, outputStream);
+            return outputStream;
+        }
+    }
+
+    @SuppressWarnings("squid:S1166")
```

Comment:
Please add a comment to the annotation that explains what is suppressed and why

Reply:
done